Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.cpp

```diff
@@ -570,7 +570,10 @@ void GraphicsContextGLANGLE::readPixelsBufferObject(IntRect rect, GCGLenum forma
 
 std::optional<IntSize> GraphicsContextGLANGLE::readPixelsImpl(IntRect rect, GCGLenum format, GCGLenum type, GCGLsizei bufSize, uint8_t* rawData, bool readingToPixelBufferObject)
 {
-    auto data = unsafeMakeSpan(rawData, checkedProduct<size_t>(rect.width(), rect.height(), 4));
+    auto product = checkedProduct<size_t>(rect.width(), rect.height(), 4);
```

Comment:
The spans are already made by the callers, and this bug was added when the unsafeMakeSpan stuff was added.

Try to organise the code so that the signature becomes:
```
std::optional<IntSize> GraphicsContextGLANGLE::readPixelsImpl(IntRect rect, GCGLenum format, GCGLenum type, std::span<uint8_t> data)
```


Try if the PBO read path can construct zero-length std::span into the offset, and detect readiongToPixelBufferObject by that property (size() == 0).